pandas/core/frame.py

```diff
@@ -5304,6 +5304,147 @@ def _construct_result(self, result) -> "DataFrame":
+
+        Parameters
+        ----------
+        other : DataFrame
+            Object to compare with.
+
+        axis : {0 or 'index', 1 or 'columns'}, default 1
+            Determine how the differences are stacked.
+            * 0, or 'index' : Stack differences on neighbouring rows.
+            * 1, or 'columns' : Stack differences on neighbouring columns.
+
+        keep_indices : bool, default False
+            Whether to keep the rows and columns that are equal, or drop them.
+
+        keep_values : bool, default False
+            Whether to keep the values that are equal, or show as NaNs.
+
+        Returns
+        -------
+        DataFrame
+            DataFrame that shows the differences stacked side by side.
+
+        See Also
+        --------
+        Series.differences: Show differences.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame(
+        ...     {
+        ...         "col1": ["a", "a", "b", "b", "a"],
+        ...         "col2": [1.0, 2.0, 3.0, np.nan, 5.0],
+        ...         "col3": [1.0, 2.0, 3.0, 4.0, 5.0]
+        ...     },
+        ...     columns=["col1", "col2", "col3"],
+        ... )
+        >>> df
+          col1  col2  col3
+        0    a   1.0   1.0
+        1    a   2.0   2.0
+        2    b   3.0   3.0
+        3    b   NaN   4.0
+        4    a   5.0   5.0
+
+        >>> df2 = df.copy()
+        >>> df2.loc[0, 'col1'] = 'c'
+        >>> df2.loc[2, 'col3'] = 4.0
+        >>> df2
+          col1  col2  col3
+        0    c   1.0   1.0
+        1    a   2.0   2.0
+        2    b   3.0   4.0
+        3    b   NaN   4.0
+        4    a   5.0   5.0
+
+        Stack the differences on columns
+
+        >>> df.differences(df2)
+          col1       col3
+          self other self other
+        0    a     c  NaN   NaN
+        2  NaN   NaN  3.0   4.0
+
+        Stack the differences on rows
+
+        >>> df.differences(df2, axis=0)
+                col1  col3
+        0 self     a   NaN
+          other    c   NaN
+        2 self   NaN   3.0
+          other  NaN   4.0
+
+        Keep all the original indices (rows and columns)
+
+        >>> df.differences(df2, keep_indices=True)
+          col1       col2       col3
+          self other self other self other
+        0    a     c  NaN   NaN  NaN   NaN
+        1  NaN   NaN  NaN   NaN  NaN   NaN
+        2  NaN   NaN  NaN   NaN  3.0   4.0
+        3  NaN   NaN  NaN   NaN  NaN   NaN
+        4  NaN   NaN  NaN   NaN  NaN   NaN
+
+        Keep all original indices and data
+
+        >>> df.differences(df2, keep_indices=True, keep_values=True)
+          col1       col2       col3
+          self other self other self other
+        0    a     c  1.0   1.0  1.0   1.0
+        1    a     a  2.0   2.0  2.0   2.0
+        2    b     b  3.0   3.0  3.0   4.0
+        3    b     b  NaN   NaN  4.0   4.0
+        4    a     a  5.0   5.0  5.0   5.0
+        """
+        from pandas.core.reshape.concat import concat
```

Comment:
Is this copy / paste for the Series method as well? If so should just place in NDFrame in pandas.core.generic

Reply:
No they are not. Both the docstring and the code are slightly different, although they share quite some common lines.

Reply:
Can you try to create a generic implementation in NDFrame? w.r.t. the docstring can use the Appender / Substitution decorators (will see other uses of them in pandas.core.generic)

Reply:
Have combined the two into NDFrame. 

I have kept the docstrings separate though, just like NDFrame.drop(), because their examples which take about two thirds of the space are completely different, and the rest are also not the same in that their input types are different, return types are different, and the explanations of arguments are also slightly different. 

I'm not sure whether it'd be good to decorate 2 Appenders thought, like maybe `Appender(generic._shared_docs["differences"])` followed by `Appender(_shared_docs["differences"])` (or maybe in reverse order?), where the former includes the input/output, then the latter includes the respective examples.

Let me know!